crates/services/producer/src/block_producer.rs

```diff
@@ -88,6 +92,60 @@ pub struct Producer<ViewProvider, TxPool, Executor, GasPriceProvider, ConsensusP
     pub consensus_parameters_provider: ConsensusProvider,
 }
 
+impl<ViewProvider, TxPool, Executor, GasPriceProvider, ConsensusProvider>
+    Producer<ViewProvider, TxPool, Executor, GasPriceProvider, ConsensusProvider>
+where
+    ViewProvider: AtomicView + 'static,
+    ViewProvider::LatestView: BlockProducerDatabase,
+    ConsensusProvider: ConsensusParametersProvider,
+{
+    pub async fn produce_and_execute_predefined(
+        &self,
+        predefined_block: &Block,
+    ) -> anyhow::Result<UncommittedResult<Changes>>
+    where
+        Executor: ports::BlockProducer<Vec<Transaction>> + 'static,
+    {
+        let _production_guard = self.lock.lock().await;
+
+        let source = predefined_block.transactions().to_vec();
+
+        let height = predefined_block.header().consensus().height;
+
+        let block_time = predefined_block.header().consensus().time;
+
+        let header = self.new_header(height, block_time).await?;
```

Comment:
This will fetch latest DA block height, but we want to use the predefined DA block height.